src/main/java/org/wildfly/security/http/HttpServerAuthenticationMechanismFactory.java

```diff
@@ -32,20 +32,20 @@
      * Get the names of the HTTP authentication mechanisms that can be supplied by this factory filtered by the supplied
      * properties.
      *
-     * @param properties the selection criteria for the mechanisms.
-     * @return the names of the supported HTTP authentication mechanisms.
+     * @param properties the {@code non-null} set of properties to pass configuration to the mechanisms that may be evaluated for mechanism availability.
```

Comment:
So in your PoV,  the fact that `ServerMechanismFactoryImpl.getMechanismNames accepts null` is not breaking the contract? Shouldn't ServerMechanismFactoryImpl.getMechanismNames validates properties if they are really non-null.